app/src/main/java/com/github/se/travelpouch/model/gps/GPSRepository.kt

```diff
@@ -0,0 +1,107 @@
+package com.github.se.travelpouch.model.gps
+
+import android.annotation.SuppressLint
+import android.location.Location
+import android.util.Log
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.maps.model.LatLng
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.map
+
+/**
+ * Repository for handling GPS location updates. This class provides a Flow for listening to
+ * continuous GPS location updates and a Task for fetching the current location.
+ *
+ * @param fusedLocationClient The FusedLocationProviderClient instance to use for location updates.
+ */
+class GPSRepository(private val fusedLocationClient: FusedLocationProviderClient) {
+
+  companion object {
+    private const val LOCATION_UPDATE_INTERVAL = 10000L // 10 seconds
+    private const val LOCATION_FASTEST_INTERVAL = 2000L // 2 seconds
+    private const val TAG = "GPSRepository"
+  }
+
+  /**
+   * Provides a Flow for listening to continuous GPS location updates. Consumers of this Flow will
+   * receive the latest location data.
+   */
+  @SuppressLint("MissingPermission") // Ensure permissions are handled elsewhere
```

Comment:
These annotations require a special attention in the next sprint when we will implement the UI linked or we will get crashes. You could use a condition to check if the permission is granted. This would however need the context.
I let you decide how you want to manage this.

Reply:
yes, I thought about doing that in the next sprint when I do the UI part. My idea is to request GPS permission when the user accesses the map